### PR TITLE
Add try/catch to async onMapInits

### PR DESCRIPTION
**Pull request by ntatko**
Resolves: https://github.com/MonsantoCo/ol-kit/issues/201

### PR Safety Checklist:

 - [ ] Added the task to the appropriate release doc under **Enhancements** or **Bug Fixes**
 - [ ] Bump `package.json` & `package-lock.json` version numbers to appropriate release
 - [ ] (optional) All external API changes have been documented
 - [ ] (optional) Build docs: `npm run docs`

### Quick Description of Changes (+ screenshots for ui changes):

Basically, any async requests with the possibility to fail can prevent any other child components from rendering. This is a quick fix to get around this, although a more better solution can be proposed.

### CODE REVIEW

**Comment by stazrad** [comment]
@ntatko should we just add a catch on the internal promise chain that will throw or log errors that occur in any `onMapInit` callback? `.catch` added here: https://github.com/MonsantoCo/ol-kit/blob/17017537228a7e7df3a407f185612eabc2e6acf4/src/Map/Map.js#L53

(maybe even a `.finally` that still renders your app even after the catch but errors in the console?)

**Comment by ntatko** [comment]
Part of the reason I'm leaning on this fix is that I don't think we can force an erroring methodology on our IAs

**Comment by stazrad** [comment]
> Part of the reason I'm leaning on this fix is that I don't think we can force an erroring methodology on our IAs

@ntatko this wouldn't be forcing anything on anyone- it's really just due diligence of error handling that we should be doing anyway- they can always catch and handle errors before it hits our layer if they want

**Comment by ntatko** [comment]
> they can always catch and handle errors before it hits our handler if they want

True. I'll whip something up, and throw it in another branch. Closing this PR in favor of that.
